frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.tsx

```diff
@@ -0,0 +1,174 @@
+import _ from "underscore";
+import { merge } from "icepick";
+import { isNotNull } from "metabase/core/utils/types";
+import { getColorsForValues } from "metabase/lib/colors/charts";
+import { formatStaticValue } from "metabase/static-viz/lib/format";
+import { ColorPalette } from "metabase/lib/colors/types";
+import {
+  ChartSettings,
+  Series,
+  SeriesWithOneOrLessDimensions,
+  SeriesWithTwoDimensions,
+} from "../../XYChart/types";
+
+export function getSeriesWithColors(
+  multipleSeries: (SeriesWithOneOrLessDimensions | SeriesWithTwoDimensions)[][],
+  settings: ChartSettings,
+  palette: ColorPalette,
+): (SeriesWithOneOrLessDimensions | SeriesWithTwoDimensions)[][] {
+  const isMultipleSeries = multipleSeries.length > 1;
+  const keys = getSeriesKeys(multipleSeries, isMultipleSeries);
+
+  const seriesColors = settings.series_settings
+    ? _.mapObject(settings.series_settings, value => {
+        return value.color;
+      })
+    : undefined;
+  const chartColors = getColorsForValues(
+    keys,
+    removeEmptyValues(seriesColors),
```

Comment:
wrap `seriesColors` inside `removeEmptyValues` to avoid having TS complained about value of the `seriesColors` map is nullable.